### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by dougnazar**
A few fixes I came across and some codeql cleanups.

### CODE REVIEW

**Comment by baudneo** [comment]
Damn bro, where did you come from? Lol.

You're really hitting the repo hard with fixes, good job and thank you!

File: src/zm_monitor.cpp [hunk 1]
```diff
@@ -2743,23 +2743,27 @@ void Monitor::TimestampImage(Image *ts_image, SystemTimePoint ts_time) const {
   while (*s_ptr && ((unsigned int)(d_ptr - label_text) < (unsigned int) sizeof(label_text))) {
     if ( *s_ptr == config.timestamp_code_char[0] ) {
       bool found_macro = false;
+      int rc = 0;
       switch ( *(s_ptr+1) ) {
         case 'N' :
-          d_ptr += snprintf(d_ptr, sizeof(label_text)-(d_ptr-label_text), "%s", name.c_str());
+          rc = snprintf(d_ptr, sizeof(label_text)-(d_ptr-label_text), "%s", name.c_str());
```
Comment (Carbenium) [inline src/zm_monitor.cpp]: Maybe pull out the `sizeof(label_text)-(d_ptr-label_text)` calculations. GCC with -O2 seems to generate shorter assembly and quick benchmarking shows a 10% improvement there.

**Review by Carbenium — APPROVED** [review]
See one comment. Otherwise LGTM.

File: src/zm_packetqueue.cpp [hunk 2]
```diff
@@ -314,7 +314,7 @@ void PacketQueue::clearPackets(const std::shared_ptr<ZMPacket> &add_packet) {
   } // end while
 
   if (keyframe_interval > pre_event_video_packet_count) {
-    Warning("Max Image Buffer setting is too low! Needs to be great than keyframe interval. Keyframe interval is at least %d", keyframe_interval);
```
Comment (connortechnology) [inline src/zm_packetqueue.cpp]: This isn't correct.  The message is correct, I suspect the use of pre_event_video_packet_count is wrong.  Please remove this and I will take a look and correct this properly.  The intention here is to detect and warn that the keyframe interval being sent from the camera is greater than the max queue size, so it is impossible for ZM to queue enough packets.  It is new code I added this just a few days ago.

**Comment by connortechnology** [comment]
Regarding the: Cast operand to larger destination size to avoid possible overflow is

I kinda hate these random casts.  I get why they might be needed, I'm just wondering if there is a nicer way to do it.

**Comment by dougnazar** [comment]
For most of them looks like an inline function would probably work, silently promoting the operands. Not sure if it would work on 32bit for the first one, since that's specifically an `unsigned long long`.

`size_t calculate_image_buffer_size(size_t width, size_t height, size_t colours);`

File: src/zm_packetqueue.cpp (hunk 2 above)
Comment (dougnazar) [reply src/zm_packetqueue.cpp]: np. I wasn't sure, but if I increased max queue size from 0 (to 120), I was getting other errors. Keyframe interval was around 60 as I remember it.

**Comment by connortechnology** [comment]
I guess what I am saying is that.... we already made imagesize an unsigned long long, presumably with the idea that that would fix the overflow problem.  Your casting here seems to suggest that it doesn't.  That we need to upgrade one of the operands to make it do 64bit multiplication before assigning to imagesize.  We can be fairly sure for at least a few more years that width and height won't be very large, but we could just make them 64 integers too.  I suppose another thought I have is that perhaps we should be using size_t instead of unsigned long long.

**Comment by dougnazar** [comment]
Right, the issue (and warning) is that the calculation happens before the promotion.
`size_t` should be fine on all platforms. that still handles 8k images on 32bit (4 bytes), and is 8 bytes on 64bit.

Using `size_t` for size, width, height, etc. should also fix the issue. If that's your preference, I can look at doing that. A quick grep shows only about 500 lines.

**Comment by Carbenium** [comment]
Wouldn't it make sense to use the fixed-width typedefs (`[u]int{8,16,32,64}`) at least in those places where we don't want this platform-dependent ambiguity? ``

**Pull request merged**